apps/meteor/app/apps/server/bridges/settings.ts

```diff
@@ -50,6 +52,39 @@ export class AppSettingBridge extends ServerSettingBridge {
 		return Boolean(setting && !setting.secret);
 	}
 
+	protected async getReadableSettingById(id: string, appId: string): Promise<ISetting | null> {
+		this.orch.debugLog(`The app ${appId} is checking if they can read the setting ${id}`);
+		const app = Apps.self?.getManager().getOneById(appId);
+		if (!app) {
+			this.orch.debugLog(`The app ${appId} is not found.`);
+			return null;
+		}
+
+		const { permissions } = app.getInfo();
+		if (!permissions) {
+			this.orch.debugLog(`The app ${appId} has no configured permissions.`);
+			return null;
+		}
+
+		const readSettingsPermission = permissions.find((perm) => perm.name === 'server-setting.read');
+		if (!readSettingsPermission) {
+			this.orch.debugLog(`The app ${appId} has no server-setting.read permission.`);
+			return null;
+		}
+
+		const readSettings = readSettingsPermission as IReadSettingPermission;
+		// If the setting is in the hiddenSettings list (defined within the permission), then it can bypass the hidden flag.
+		// If not, then it must be a non-hidden setting. This is to allow apps to read hidden settings if they have the permission to do so.
+		const setting = readSettings.hiddenSettings?.includes(id) ? await Settings.findOneById(id) : await Settings.findOneNotHiddenById(id);
```

Comment:
Should we log somehow an app attempted to read a setting it doesnt have access to instead of the setting "not being found"?

Reply:
I think we could log via orch.debugLog (not sure if that valuable though), but I don't think we should return this to the end user.

I think we have enough tracing to troubleshoot most issues that might arise, wdyt?

Reply:
I'm not sure... what would we do with this info?

Reply:
I can see it being useful as an app developer who forgot to add that specific setting to the permisson list, but I mean, it should be clear from the logs that no setting was found (also, not sure if this should be a widely documented feature of apps engine)